.gitignore

```diff
@@ -1,10 +1,29 @@
-*~ bak .bak  *.bak
-config.cache config.status config.log
+*~
+*.bak
+*/.deps
+*/.libs
 FcgiBin
-Makefile Makefile.in
-aclocal.m4 configure
-fcgi_config.h fcgi_config.h.in
+Makefile*
+aclocal.m4
+autom4te.cache/
+build*
+cgi-fcgi/cgi-fcgi
+compile
+config.*
+configure
+depcomp
+examples/authorizer
```

Comment:
why not add this in .gitignore of examples directory ?

Reply:
Probably because this is a git, but not a subversion...
More hidden files == less transparency.

Reply:
I even didn't look for hidden files in subdirs, because I have not expected to find them there...

Reply:
ok but original one use ignore file in each directory.
i think is goot to keep all that specific to that directory in each .gitignore and all that are generic in top .gitignore ?!

Reply:
> ok but original one use ignore file in each directory.

Original project was in svn or even in cvs. (I don't remember now.)
We do not have to keep such archaic structure of source tree...

> i think is goot to keep all that specific to that directory in each .gitignore and all that are generic in top .gitignore ?!

I disagree with you here. I am using git since 2009 and [have contributed](https://www.openhub.net/accounts/tehnick) to many projects and this is the first time when I have faced with `.gitignore` files in sub-directories. Could you write any real pros for this use case? Could you show examples of such using of `.gitignore` in other projects?